lib/views/frontend/spree/orders/_stripe_payment_request_button.html.erb

```diff
@@ -0,0 +1,48 @@
+<% if current_order.present? && cart_checkout_payment_method.present? %>
+  <div id="payment-request">
+    <%= form_tag update_checkout_path(:payment), method: :patch do %>
```

Comment:
Is this form really needed? I think it could be better to fully handle this via JS since this partial could be added inside another form and this could lead to issues.

Reply:
It would imply modifying some JS code. At the moment, as much JS as possible is shared between the Elements, Payment Intents, and cart checkout button functionalities.

The first two implementations use the form in order to collect customer's data and submit payment information to the application, in this case this is not strictly necessary as the browser is doing it, but by using a form that behaves like the other twos we get to retain the same behavior and reuse the JS code. 

I can try to remove it so we can compare the results, what do you think?

Reply:
If we can I would give it a try since nested forms are not allowed for valid HTML.